src/aoptk/abbreviation_translator_dictionary.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+import re
+from aoptk.abbreviation_translator import AbbreviationTranslator
+
+
+class AbbreviationTranslatorDictionary(AbbreviationTranslator):
+    """Translates abbreviations in text using a provided dictionary."""
+
+    def __init__(self, pdf_dictionary: dict[str, str]):
+        self.pdf_dictionary = pdf_dictionary
+
+    def translate_abbreviation(self, text: str) -> str:
+        """Translate abbreviations in the text using the provided dictionary."""
+        translated_text = text
+
+        for abbreviation, full_form in self.pdf_dictionary.items():
+            patterns = [
+                (r"\b" + re.escape(abbreviation) + r"s\b", full_form + "s"),
+                (r"\b" + re.escape(abbreviation) + r"\b", full_form),
+            ]
+            for pattern, replacement in patterns:
+                matches = list(re.finditer(pattern, translated_text, flags=re.IGNORECASE))
+                translated_text = self.fix_capitalization(translated_text, replacement, matches)
+        return translated_text
+
+    def fix_capitalization(
```

Comment:
I think this function would also deserve a dedicated test case, as it is somewhat non-obvious what it does and whether it really works.